pysumma/preprocessing/create_local_attribute.py

```diff
@@ -0,0 +1,237 @@
+    def save_local_attribute(self, df_format, csv_name):
+        """[summary]
+
+        Args:
+            df_format ([type]): [description]
+            csv_name ([type]): [description]
+
+        Returns:
+            [type]: [description]
+        """
+        local_attribute = df_format.dropna()
+        local_attribute.to_csv(csv_name, index=False)
+        return local_attribute
+
+    def one_col_add(self, df_format, col_name, value=10):
+        df_format[col_name] = value
+        return df_format
+
+    def down_hru_index(self, df_format, fdr, hru_id_col_name="hruId", downhruindex_col_name="downHRUindex"):
+        hruid_array = df_format[hru_id_col_name].values.reshape(
+            fdr.shape[1], fdr.shape[0])
+        hruid = pd.DataFrame(hruid_array)
+        hruid = hruid.T
+        downHRUindex = pd.DataFrame(np.zeros((hruid.shape[0], hruid.shape[1])))
+        downHRUindex.astype('int64').head()
+        for index_r in range(fdr.shape[0]):
+            for index_c in range(fdr.shape[1]):
+                if fdr.loc[index_r, index_c] == 1 and index_r-1 >= 0 and index_c+1 >= 0:
+                    downHRUindex.loc[index_r,
+                                     index_c] = hruid.loc[index_r-1, index_c+1]
+                elif fdr.loc[index_r, index_c] == 2 and index_r-1 >= 0 and index_c >= 0:
+                    downHRUindex.loc[index_r,
+                                     index_c] = hruid.loc[index_r-1, index_c]
+                elif fdr.loc[index_r, index_c] == 3 and index_r-1 >= 0 and index_c-1 >= 0:
+                    downHRUindex.loc[index_r,
+                                     index_c] = hruid.loc[index_r-1, index_c-1]
+                elif fdr.loc[index_r, index_c] == 4 and index_r >= 0 and index_c-1 >= 0:
+                    downHRUindex.loc[index_r,
+                                     index_c] = hruid.loc[index_r, index_c-1]
+                elif fdr.loc[index_r, index_c] == 5 and index_r+1 >= 0 and index_c-1 >= 0:
+                    downHRUindex.loc[index_r,
+                                     index_c] = hruid.loc[index_r+1, index_c-1]
+                elif fdr.loc[index_r, index_c] == 6 and index_r+1 >= 0 and index_c >= 0:
+                    downHRUindex.loc[index_r,
+                                     index_c] = hruid.loc[index_r+1, index_c]
+                elif fdr.loc[index_r, index_c] == 7 and index_r+1 >= 0 and index_c+1 >= 0:
+                    downHRUindex.loc[index_r,
+                                     index_c] = hruid.loc[index_r+1, index_c+1]
+                elif fdr.loc[index_r, index_c] == 8 and index_r >= 0 and index_c+1 >= 0:
+                    downHRUindex.loc[index_r,
+                                     index_c] = hruid.loc[index_r, index_c+1]
+                elif fdr.loc[index_r, index_c] < 0 and index_r >= 0 and index_c+1 >= 0:
+                    downHRUindex.loc[index_r, index_c] = -9999
+                else:
+                    downHRUindex.loc[index_r, index_c] = 0
```

Comment:
I have no idea what any of this means - could you provide either some comments about what these are or rename some of the variables/raw integers so that this is easier to understand?